test/fuelChainConsensus.ts

```diff
@@ -137,53 +147,56 @@ describe('Fuel Chain Consensus', async () => {
         let signer1: string;
+        let signer2: string;
         before(async () => {
             signer1 = env.addresses[1];
+            signer2 = env.addresses[2];
         });
 
-        it('Should be able to set authority as owner', async () => {
-            expect(await env.fuelChainConsensus.authorityKey()).to.not.be.equal(signer1);
+        it('Should be able to set comitter as admin', async () => {
+            expect(await env.fuelChainConsensus.hasRole(committerRole, signer1)).to.equal(false);
 
-            // Set authority
-            await expect(env.fuelChainConsensus.setAuthorityKey(signer1)).to.not.be.reverted;
-            expect(await env.fuelChainConsensus.authorityKey()).to.be.equal(signer1);
+            // Set comitter
+            await expect(env.fuelChainConsensus.grantRole(committerRole, signer1)).to.not.be.reverted;
+            expect(await env.fuelChainConsensus.hasRole(committerRole, signer1)).to.equal(true);
         });
 
-        it('Should not be able to set authority as non-owner', async () => {
-            expect(await env.fuelChainConsensus.authorityKey()).to.be.equal(signer1);
+        it('Should not be able to set committer role as non-admin', async () => {
+            expect(await env.fuelChainConsensus.hasRole(committerRole, signer2)).to.equal(false);
 
-            // Attempt set authority
+            // Attempt set comitter
             await expect(
-                env.fuelChainConsensus.connect(env.signers[1]).setAuthorityKey(env.poaSignerAddress)
+                env.fuelChainConsensus.connect(env.signers[1]).grantRole(committerRole, signer2)
             ).to.be.revertedWith(
                 `AccessControl: account ${env.addresses[1].toLowerCase()} is missing role ${defaultAdminRole}`
             );
-            expect(await env.fuelChainConsensus.authorityKey()).to.be.equal(signer1);
+            expect(await env.fuelChainConsensus.hasRole(committerRole, signer2)).to.equal(false);
         });
 
-        it('Should be able to switch authority back', async () => {
-            expect(await env.fuelChainConsensus.authorityKey()).to.not.be.equal(env.poaSignerAddress);
+        it('Should not be able to make commits as non-comitter', async () => {
+            await expect(env.fuelChainConsensus.connect(env.signers[2]).commit(randomBytes32(), 9)).to.be.revertedWith(
+                `AccessControl: account ${env.addresses[2].toLowerCase()} is missing role ${committerRole}`
+            );
+        });
 
-            // Set authority
-            await expect(env.fuelChainConsensus.setAuthorityKey(env.poaSignerAddress)).to.not.be.reverted;
-            expect(await env.fuelChainConsensus.authorityKey()).to.be.equal(env.poaSignerAddress);
+        it('Should be able to make commits as comitter', async () => {
+            await expect(env.fuelChainConsensus.connect(env.signers[1]).commit(randomBytes32(), 9)).to.not.be.reverted;
```

Comment:
It would be better if we could retrieve whatever is committed to prove that they have actually committed and not that the call has not reverted.

Reply:
Good idea. I should at least add a getter for manually analysis of the commit slots